sky/utils/kubernetes/deploy_remote_cluster.sh

```diff
@@ -163,7 +199,7 @@ run_remote "$HEAD_NODE" "
             sleep 5
         fi
     done
-    if [ $i -eq 3 ]; then
+    if [ \$i -eq 3 ]; then
```

Comment:
Why do we need this?

Reply:
The commands are to be executed in remote server, without `\` it will try to catch the `$i` var in this script.

Reply:
Good catch!
minor: if we use `'` for quoting this script, it should be fine to use `$` instead of `\$`?

Reply:
Yes, but we also need to catch the variable in the script here, e.g. PASSWORD, K3S_TOKEN, etc.